components/home/reducers.js

```diff
@@ -0,0 +1,53 @@
+import { UPDATE_REACTION, REQUEST_UPLOAD_LIST } from './actionTypes';
+import { INITIAL_STATE, USER_UPLOADS } from './constants';
+
+/**
+ * Creates a Javascript Map with the user uploads mapped by id
+ *
+ * @param {Array} USER_UPLOADS - a users uploads
+ * @return {Map} - the user uploads
+ */
+
+function generateUploadsMap() {
+    const uploads = new Map();
+
+    USER_UPLOADS.forEach(userUpload => {
+        const { id } = userUpload;
+
+        uploads.set(id, userUpload);
+    });
+
+    return uploads;
+}
+
+function updateUploadReaction(id, reaction, uploads) {
```

Comment:
this should be `function updateUploadReaction(id, type, uploads) {...}`

Reply:
Alright, that makes more sense